pkg/jobs/update.go

```diff
@@ -139,6 +139,12 @@ func (j *Job) Update(ctx context.Context, txn *kv.Txn, updateFn UpdateFn) error
 	return j.update(ctx, txn, useReadLock, updateFn)
 }
 
+// UpdateWithLock is similar to Update, but locks the row to prevent race conditions during concurrent update attempts
+func (j *Job) UpdateWithLock(ctx context.Context, txn *kv.Txn, updateFn UpdateFn) error {
+	const useReadLock = true
+	return j.update(ctx, txn, useReadLock, updateFn)
+}
+
```

Comment:
It would be nice if changes like this (to the jobs system) were pulled into separate change (might be part of the same PR).  At the very least, this should be reviewed by the jobs team (@dt ); Things like this always raise questions if the caller (in this case TTL job) could hold this txn open for long periods of time.  If so, this is not just 1 row that's getting locked up -- but every job related command, including show jobs and show changefeed jobs.

Reply:
hmm, not sure this is needed now that we're only updating one row.
sorry i missed that we used this here.

Reply:
Reverted this back to how it was.